network/p2p/gossip/gossip.go

```diff
@@ -323,9 +327,10 @@ type PushGossiper[T Gossipable] struct {
 }
 
 type BranchingFactor struct {
-	Validators    int
-	NonValidators int
-	Peers         int
+	StakePercentage float64
```

Comment:
nit: wonder if it makes sense to take the `num` strategy here like we do for warp?

Reply:
I suppose we don't need to because we don't require determinism.